### PR TITLE
added generated code for AWS

### PR DESCRIPTION
**Pull request by xh3b4sd**

### CODE REVIEW

File: pkg/apis/cluster/v1alpha1/aws_types.go [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type AWS struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+	Spec              AWSSpec `json:"spec"`
+}
+
+type AWSSpec struct {
+	Cluster       AWSSpecCluster       `json:"cluster" yaml:"cluster"`
```
Comment (xh3b4sd) [inline pkg/apis/cluster/v1alpha1/aws_types.go]: The cluster block is basically copied from the KVM structures and then renamed accordingly.

File: pkg/apis/cluster/v1alpha1/aws_types.go [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+
+type AWSSpecClusterKubernetesIngressController struct {
+	Docker         AWSSpecClusterKubernetesIngressControllerDocker `json:"docker" yaml:"docker"`
+	Domain         string                                          `json:"domain" yaml:"domain"`
+	WildcardDomain string                                          `json:"wildcardDomain" yaml:"wildcardDomain"`
+	InsecurePort   int                                             `json:"insecurePort" yaml:"insecurePort"`
+	SecurePort     int                                             `json:"securePort" yaml:"securePort"`
+}
+
+type AWSSpecClusterKubernetesIngressControllerDocker struct {
```
Comment (xh3b4sd) [inline pkg/apis/cluster/v1alpha1/aws_types.go]: Yeah, I know. I talked with Pawel about this already. We agreed on making progress now and coming up with a scheme meanwhile. This is nothing that should block us now. So lets start a discussion and document contribution guidelines for this repo asynchronously. For now I just want to get shit done and don't want to get blocked by naming. I am glad to help fixing the naming here later together.

File: pkg/apis/cluster/v1alpha1/aws_types.go [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+type AWSSpecClusterNode struct {
+	ID string `json:"id" yaml:"id"`
+}
+
+type AWSSpecClusterVault struct {
+	Address string `json:"address" yaml:"address"`
+	Token   string `json:"token" yaml:"token"`
+}
+
+type AWSSpecAWS struct {
```
Comment (xh3b4sd) [inline pkg/apis/cluster/v1alpha1/aws_types.go]: I just applied all the information from the TPR repo.

File: pkg/apis/cluster/v1alpha1/aws_types.go [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+	API     AWSSpecAWSAPI     `json:"api" yaml:"api"`
+	AZ      string            `json:"az" yaml:"az"`
+	Etcd    AWSSpecAWSEtcd    `json:"etcd" yaml:"etcd"`
```
Comment (xh3b4sd) [inline pkg/apis/cluster/v1alpha1/aws_types.go]: I restructured that a bit in a way that made more sense to me. The amount of information stays the same though. Just the structure changed a bit.

File: pkg/clientset/versioned/typed/cluster/v1alpha1/aws.go [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+		ns:     namespace,
+	}
+}
+
+// Get takes name of the aWS, and returns the corresponding aWS object, and an error if there is any.
+func (c *aWSs) Get(name string, options v1.GetOptions) (result *v1alpha1.AWS, err error) {
+	result = &v1alpha1.AWS{}
+	err = c.client.Get().
+		Namespace(c.ns).
+		Resource("awss").
```
Comment (rossf7) [inline pkg/clientset/versioned/typed/cluster/v1alpha1/aws.go]: The pluralisation here is horrible due to TPRs. Will this affect us? Or do we just use the autogenerated client and it does the magic?

File: pkg/apis/cluster/v1alpha1/aws_types.go (hunk 2 above)
Comment (rossf7) [reply pkg/apis/cluster/v1alpha1/aws_types.go]: That is a hell of a long name but I'm OK with fixing this later if switching to the auto generated client helps us now.

File: pkg/clientset/versioned/typed/cluster/v1alpha1/aws.go (hunk 5 above)
Comment (kopiczko) [reply pkg/clientset/versioned/typed/cluster/v1alpha1/aws.go]: I don't know this either. This is something to figure out. In the worst case scenario we'll have to add some sed calls or just live with that. I'd say we take it with the generation goodness if necessary.

**Pull request merged**